rocks-bin/src/pin.rs

```diff
@@ -0,0 +1,24 @@
+use clap::Args;
+use eyre::eyre;
+use eyre::Result;
+use rocks_lib::operations;
+use rocks_lib::{
+    config::{Config, LuaVersion},
+    package::PackageReq,
+    tree::Tree,
+};
+
+#[derive(Args)]
+pub struct Pin {
+    package: PackageReq,
+}
+
+pub fn pin(data: Pin, config: Config) -> Result<()> {
+    let tree = Tree::new(config.tree().clone(), LuaVersion::from(&config)?)?;
+
+    if let Some(mut rock) = tree.has_rock(&data.package) {
```

Comment:
question: wouldn't you want to pin an exact `Package` instead of a `PackageReq`?

Otherwise, what if there are multiple rocks installed that match the `PackageReq`?